include/deal.II/cgal/utilities.h

```diff
@@ -103,6 +114,44 @@ namespace CGALWrappers
   cgal_surface_mesh_to_cgal_coarse_triangulation(
     CGAL::Surface_mesh<typename C3t3::Point::Point> &surface_mesh,
     C3t3 &                                           triangulation);
+
+  /**
+   * Given two triangulated surface meshes, the corefinement operation consists
+   * in refining both meshes so that their intersection polylines are a subset
+   * of edges in both refined meshes.
+   *
+   * The corefinement of two triangulated surface meshes can naturally be used
+   * for computing Boolean operations on the volumes bounded by surfaces,
+   * according to the chosen BooleanOperation. See BooleanOperation for a list
+   * of available operations. As an example consider the following case with a
+   * cube and the green polyhedron.
+   *
+   * @image html hedra_cube.png
+   *
+   * The shaded regions in the following show the results of the intersection,
+   * union, difference and corefinement between a cube and the green polyhedron.
+   *
+   * @image html boolean_intersection.png
+   * @image html boolean_union.png
+   * @image html boolean_difference.png
+   * @image html corefinement.png
+   *
+   * See the CGAL documentation for an extended discussion and several examples:
+   * https://doc.cgal.org/latest/Polygon_mesh_processing/index.html#title14
+   *
+   * @param[in] surf_1 The first surface mesh.
+   * @param[in] surf_2 The second surface mesh.
+   * @param[out] outsurf The surface mesh with storing the result of the boolean
+   * operation. Notice that in case of corefinement only, the corefined meshes
+   * will be the first ones.
+   * @param bool_op See BooleanOperation for the list of the allowed operations.
+   */
+  template <typename CGALPointType>
+  void
+  compute_boolean_operation(CGAL::Surface_mesh<CGALPointType> &surf_1,
+                            CGAL::Surface_mesh<CGALPointType> &surf_2,
+                            CGAL::Surface_mesh<CGALPointType> &outsurf,
+                            const BooleanOperation &           bool_op);
```

Comment:
If you are not providing a default boolean operation as argument, then the order of the arguments of the function should be:
1. surface_mesh_1
2. surface_mesh_2
3. boolean_operation
4. output_surface_mesh
to follow deal.II coding standards: all const (input) arguments first, then all non const (output) arguments, then all const optional arguments with a default value.

Please, spell out the names entirely.